### PR TITLE
conditional routing headers for alb

### PR DESCRIPTION
**Pull request by thisismana**

### CODE REVIEW

File: .terraform.lock.hcl [hunk 1]
```diff
@@ -2,19 +2,20 @@
 # Manual edits may be lost in future updates.
 
 provider "registry.terraform.io/hashicorp/aws" {
```
Comment (moritzzimmer) [inline .terraform.lock.hcl:4]: not sure if it makes sense to have this file under version control at all for modules. we probbaly can ignore this in git

File: alb.tf [hunk 2]
```diff
@@ -1,154 +1,254 @@
...
-  count             = var.alb_attach_public_target_group ? 1 : 0
-  load_balancer_arn = data.aws_lb.public[count.index].arn
-  port              = 80
-}
+  name        = lookup(var.target_groups[count.index], "name", null)
+  name_prefix = lookup(var.target_groups[count.index], "name_prefix", null)
 
-resource "aws_alb_target_group" "public" {
-  count = var.alb_attach_public_target_group ? 1 : 0
+  vpc_id           = data.aws_vpc.selected.id
```
Comment (moritzzimmer) [inline alb.tf:15]: does it makes sense to refactor this to a required variable as well instead of hardcoding this to `main`?

File: docs/part1.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # AWS Fargate ECS Terraform Module
```
Comment (moritzzimmer) [inline docs/part1.md:1]: guess we need to update the readme here concerning all required data source and the removed DNS stuff

File: alb.tf [hunk 4]
```diff
@@ -1,154 +1,254 @@
+/**
+* https://github.com/terraform-aws-modules/terraform-aws-alb
+*/
+
 data "aws_lb" "public" {
-  count = var.alb_attach_public_target_group ? 1 : 0
-  name  = "public"
+  name = "public"
```
Comment (moritzzimmer) [inline alb.tf:6]: does it makes sense to refactor this to a required variable as well instead of hardcoding this to `public`?

**Review by moritzzimmer — CHANGES_REQUESTED** [review]
Some minor comments/discussions.

Plus:

- run `make documentation`
- a terraformable example would be helpful to see the client side of those new enhancements

File: alb.tf (hunk 4 above)
Comment (thisismana) [reply alb.tf:6]: unused.

File: alb.tf (hunk 2 above)
Comment (thisismana) [reply alb.tf:15]: 👍

File: variables.tf [hunk 5]
```diff
@@ -24,45 +24,27 @@ variable "service_name" {
   type        = string
 }
 
+variable "vpc_id" {
```
Comment (moritzzimmer) [inline variables.tf:27]: could u move this variable down to optional region for vars?

**Review by moritzzimmer — APPROVED** [review]

**Pull request merged**